# Kickstarter SQL project
made by Blazej Kloc

For this project, I'll take on the role of a data analyst at a startup. The product team is considering launching a campaign on Kickstarter to test the viability of some offerings. I've been asked to pull data that will help the team understand what might influence the success of a campaign.

## Questions:


### What types of projects are most likely to be successful?
### Which projects fail?

## Data

 - `ID:` Kickstarter project ID
 - `name:` Name of project
 - `category:` Category of project
 - `main_category:` Main category of project
 - `goal:` Fundraising goal
 - `pledged:` Amount pledged
 - `state`: State of project (successful, canceled, etc.)
 - `backers:` Number of project backers

 source (31.10.24): https://www.kaggle.com/datasets/kemical/kickstarter-projects?resource=download


## preparing the data

In [18]:
%load_ext sql

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


In [19]:
import pandas as pd
import sqlite3

# import CSV
df = pd.read_csv("ks-projects-201612.csv", encoding='ISO-8859-1')

# connect with SQL DB
conn = sqlite3.connect("ks-project.db")

# save table SQL
df.to_sql("ksprojects", conn, if_exists="replace", index=False)

<ipython-input-19-55d36e613ff3>:5: DtypeWarning: Columns (13,14,15) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv("ks-projects-201612.csv", encoding='ISO-8859-1')


323750

In [20]:
%sql sqlite:///ks-project.db

## Quick look at the data

In [21]:
%%sql
SELECT *
  FROM ksprojects
 LIMIT 10;

 * sqlite:///ks-project.db
   sqlite:///ks-project_clean.db
Done.


[(1000002330, 'The Songs of Adelaide & Abullah', 'Poetry', 'Publishing', 'GBP', '2015-10-09 11:36:00', '1000', '2015-08-11 12:12:28', '0', 'failed', '0', 'GB', '0', None, None, None, None),
 (1000004038, 'Where is Hank?', 'Narrative Film', 'Film & Video', 'USD', '2013-02-26 00:20:50', '45000', '2013-01-12 00:20:50', '220', 'failed', '3', 'US', '220', None, None, None, None),
 (1000007540, 'ToshiCapital Rekordz Needs Help to Complete Album', 'Music', 'Music', 'USD', '2012-04-16 04:24:11', '5000', '2012-03-17 03:24:11', '1', 'failed', '1', 'US', '1', None, None, None, None),
 (1000011046, 'Community Film Project: The Art of Neighborhood Filmmaking', 'Film & Video', 'Film & Video', 'USD', '2015-08-29 01:00:00', '19500', '2015-07-04 08:35:03', '1283', 'canceled', '14', 'US', '1283', None, None, None, None),
 (1000014025, 'Monarch Espresso Bar', 'Restaurants', 'Food', 'USD', '2016-04-01 13:38:27', '50000', '2016-02-26 13:38:27', '52375', 'successful', '224', 'US', '52375', None, None, None, None),
 (1000023410, 'Support Solar Roasted Coffee & Green Energy!  SolarCoffee.co', 'Food', 'Food', 'USD', '2014-12-21 18:30:44', '1000', '2014-12-01 18:30:44', '1205', 'successful', '16', 'US', '1205', None, None, None, None),
 (1000030581, 'Chaser Strips. Our Strips make Shots their B*tch!', 'Drinks', 'Food', 'USD', '2016-03-17 19:05:12', '25000', '2016-02-01 20:05:12', '453', 'failed', '40', 'US', '453', None, None, None, None),
 (1000034518, 'SPIN - Premium Retractable In-Ear Headphones with Mic', 'Product Design', 'Design', 'USD', '2014-05-29 18:14:43', '125000', '2014-04-24 18:14:43', '8233', 'canceled', '58', 'US', '8233', None, None, None, None),
 (100004195, 'STUDIO IN THE SKY - A Documentary Feature Film (Canceled)', 'Documentary', 'Film & Video', 'USD', '2014-08-10 21:55:48', '65000', '2014-07-11 21:55:48', '6240.57', 'canceled', '43', 'US', '6240.57', None, None, None, None),
 (100004721, 'Of Jesus and Madmen', 'Nonfiction', 'Publishing', 'CAD', '2013-10-09 18:19:37', '2500', '2013-09-09 18:19:37', '0', 'failed', '0', 'CA', '0', None, None, None, None)]

In [22]:
%%sql
PRAGMA table_info(ksprojects);

 * sqlite:///ks-project.db
   sqlite:///ks-project_clean.db
Done.


[(0, 'ID ', 'INTEGER', 0, None, 0),
 (1, 'name ', 'TEXT', 0, None, 0),
 (2, 'category ', 'TEXT', 0, None, 0),
 (3, 'main_category ', 'TEXT', 0, None, 0),
 (4, 'currency ', 'TEXT', 0, None, 0),
 (5, 'deadline ', 'TEXT', 0, None, 0),
 (6, 'goal ', 'TEXT', 0, None, 0),
 (7, 'launched ', 'TEXT', 0, None, 0),
 (8, 'pledged ', 'TEXT', 0, None, 0),
 (9, 'state ', 'TEXT', 0, None, 0),
 (10, 'backers ', 'TEXT', 0, None, 0),
 (11, 'country ', 'TEXT', 0, None, 0),
 (12, 'usd pledged ', 'TEXT', 0, None, 0),
 (13, 'Unnamed: 13', 'TEXT', 0, None, 0),
 (14, 'Unnamed: 14', 'TEXT', 0, None, 0),
 (15, 'Unnamed: 15', 'TEXT', 0, None, 0),
 (16, 'Unnamed: 16', 'REAL', 0, None, 0)]

In [23]:
%%sql
SELECT category
  FROM ksprojects
LIMIT 10;

 * sqlite:///ks-project.db
   sqlite:///ks-project_clean.db
(sqlite3.OperationalError) no such column: category
[SQL: SELECT category
  FROM ksprojects
LIMIT 10;]
(Background on this error at: https://sqlalche.me/e/20/e3q8)


## Here I had problem with geting into specific categories, after a while I discover that there are spaces in column names.

In [24]:
%%sql
SELECT "category "
  FROM ksprojects
LIMIT 10;

 * sqlite:///ks-project.db
   sqlite:///ks-project_clean.db
Done.


[('Poetry',),
 ('Narrative Film',),
 ('Music',),
 ('Film & Video',),
 ('Restaurants',),
 ('Food',),
 ('Drinks',),
 ('Product Design',),
 ('Documentary',),
 ('Nonfiction',)]

## Updating data base the dataset

In [25]:
import pandas as pd
import sqlite3

# original CSV
df = pd.read_csv("ks-projects-201612.csv", encoding='ISO-8859-1')
df.columns = df.columns.str.strip()  # Delating spaces

# saving clean CSV file
df.to_csv("ks-projects-201612_clean.csv", index=False)

# CSV to .db file
conn = sqlite3.connect("ks-project_clean.db")
df_clean = pd.read_csv("ks-projects-201612_clean.csv", encoding='ISO-8859-1')
df_clean.to_sql("ksprojects2", conn, if_exists="replace", index=False) #renaming table to "ksprojects2"


conn.close()


<ipython-input-25-8414a0f25ffa>:5: DtypeWarning: Columns (13,14,15) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv("ks-projects-201612.csv", encoding='ISO-8859-1')
<ipython-input-25-8414a0f25ffa>:13: DtypeWarning: Columns (13,14,15) have mixed types. Specify dtype option on import or set low_memory=False.
  df_clean = pd.read_csv("ks-projects-201612_clean.csv", encoding='ISO-8859-1')


In [26]:
%sql sqlite:///ks-project_clean.db

In [27]:
%%sql
PRAGMA table_info(ksprojects2);

   sqlite:///ks-project.db
 * sqlite:///ks-project_clean.db
Done.


[(0, 'ID', 'INTEGER', 0, None, 0),
 (1, 'name', 'TEXT', 0, None, 0),
 (2, 'category', 'TEXT', 0, None, 0),
 (3, 'main_category', 'TEXT', 0, None, 0),
 (4, 'currency', 'TEXT', 0, None, 0),
 (5, 'deadline', 'TEXT', 0, None, 0),
 (6, 'goal', 'TEXT', 0, None, 0),
 (7, 'launched', 'TEXT', 0, None, 0),
 (8, 'pledged', 'TEXT', 0, None, 0),
 (9, 'state', 'TEXT', 0, None, 0),
 (10, 'backers', 'TEXT', 0, None, 0),
 (11, 'country', 'TEXT', 0, None, 0),
 (12, 'usd pledged', 'TEXT', 0, None, 0),
 (13, 'Unnamed: 13', 'TEXT', 0, None, 0),
 (14, 'Unnamed: 14', 'TEXT', 0, None, 0),
 (15, 'Unnamed: 15', 'TEXT', 0, None, 0),
 (16, 'Unnamed: 16', 'REAL', 0, None, 0)]

In [28]:
%%sql
SELECT name FROM sqlite_master WHERE type='table';

   sqlite:///ks-project.db
 * sqlite:///ks-project_clean.db
Done.


[('ksprojects2',)]

## Trying to answer

In [29]:
%%sql
  SELECT main_category, backers, pledged, goal,
         pledged / goal AS pct_pledged,
         (CASE
          WHEN pledged / goal > 1 THEN "Fully funded"
          WHEN pledged / goal > 0.75 THEN "Nearly funded"
          WHEN pledged / goal <= 0.75 THEN "Not nearly funded"
         END) AS funding_status
    FROM ksprojects2
   WHERE state IN ('failed')
     AND backers >= 100 AND pledged >= 20000
ORDER BY main_category, pct_pledged DESC
   LIMIT 20;

   sqlite:///ks-project.db
 * sqlite:///ks-project_clean.db
Done.


[('Art', '97', '7271', '7654.32', 0.9499210903123988, 'Nearly funded'),
 ('Art', '86', '2293.5', '2500', 0.9174, 'Nearly funded'),
 ('Art', '294', '52565.25', '60000', 0.8760875, 'Nearly funded'),
 ('Art', '22', '860', '999.99', 0.8600086000860009, 'Nearly funded'),
 ('Art', '82', '5011.9', '6000', 0.8353166666666666, 'Nearly funded'),
 ('Art', '6', '83.11', '100', 0.8311, 'Nearly funded'),
 ('Art', '57', '2440.11', '3000', 0.81337, 'Nearly funded'),
 ('Art', '185', '27630.24', '35000', 0.7894354285714286, 'Nearly funded'),
 ('Art', '126', '2585.33', '3300', 0.7834333333333333, 'Nearly funded'),
 ('Art', '47', '8549.8', '11000', 0.7772545454545454, 'Nearly funded'),
 ('Art', '552', '37621.97', '50000', 0.7524394, 'Nearly funded'),
 ('Art', '22', '898.34', '1200', 0.7486166666666667, 'Not nearly funded'),
 ('Art', '25', '732.77', '1000', 0.73277, 'Not nearly funded'),
 ('Art', '173', '9040.5', '12600', 0.7175, 'Not nearly funded'),
 ('Art', '45', '2601.01', '4000', 0.6502525, 'Not nearly funded'),
 ('Art', '16', '626.91', '1000', 0.62691, 'Not nearly funded'),
 ('Art', '34', '625.5', '1000', 0.6255, 'Not nearly funded'),
 ('Art', '13', '308.34', '500', 0.6166799999999999, 'Not nearly funded'),
 ('Art', '49', '2774.01', '4500', 0.6164466666666667, 'Not nearly funded'),
 ('Art', '27', '808.17', '1313', 0.6155140898705255, 'Not nearly funded')]

## Key Insight:
Despite having a significant number of backers and a large total pledged amount, none of these projects reached their respective funding goals, indicating a common issue where projects may have garnered interest but failed to achieve the necessary financial support to be considered successful.
This analysis highlights that projects with substantial pledges but that do not meet their goals are classified as failures, which is a crucial insight for understanding Kickstarter dynamics.

In [30]:
%%sql
SELECT main_category,
       COUNT(*) AS total_successful_projects,
       AVG(pledged) AS average_pledged,
       AVG(goal) AS average_goal,
       AVG(pledged / goal) AS avg_pct_pledged
 FROM ksprojects2
WHERE state = 'successful'
GROUP BY main_category
ORDER BY total_successful_projects DESC;

   sqlite:///ks-project.db
 * sqlite:///ks-project_clean.db
Done.


[('Music', 21763, 7190.293610255917, 5633.310235261681, 11.948693060925129),
 ('Film & Video', 21404, 13851.514675294344, 11081.275109792565, 3.2161248573526295),
 ('Publishing', 10255, 9295.826817162342, 6008.886572403706, 4.924243876704187),
 ('Art', 9654, 6964.88831365234, 4525.464167184587, 6.192172422396589),
 ('Games', 9385, 54874.61819499186, 16245.941928609484, 21.53186114499116),
 ('Design', 7959, 63489.18469028772, 16564.468714662646, 4.864323648972836),
 ('Theater', 5990, 5855.145542570948, 5072.10265275459, 1.505941597694269),
 ('Food', 5256, 17020.38314497715, 11674.14539764079, 3.105081262243008),
 ('Technology', 5062, 91117.21084946652, 27185.165679573296, 14.390363731356668),
 ('Comics', 4499, 12160.200891309174, 5666.02578128473, 13.50459668240766),
 ('Fashion', 4310, 22365.453756380513, 10112.758120649652, 3.3934941376249603),
 ('Photography', 2903, 9327.330423699626, 6317.16652084051, 1.4468953671970892),
 ('Dance', 2101, 5086.290918610181, 4489.682584483579, 1.1146361593488399),
 ('Crafts', 1671, 5615.884195092761, 2975.721723518851, 11.338724856749089),
 ('Journalism', 869, 11743.919551208288, 8960.03501726122, 1.3112648530668944)]

## Top Categories by Successful Projects:

 - Music: 21,763 successful projects
 - Film & Video: 21,404 successful projects
 - Publishing: 10,255 successful projects

 Music and Film & Video categories have the highest number of successful projects. This suggests that these categories attract a large number of backers and resonate well with the Kickstarter audience. The large volume indicates a potentially thriving community or market interest.


## Recommendations
Focus on Engagement: Projects in categories with high successful counts, such as Music and Film & Video, should continue to leverage community engagement strategies to maintain momentum.

Evaluate Goals: Categories with high average pledges but lower successful projects should reassess their funding goals to ensure they are realistic and achievable.

Explore Underrepresented Areas: There is an opportunity to cultivate support for underrepresented categories like Dance and Journalism, potentially by developing specific initiatives or community programs to attract interest